plaso/parsers/plist_plugins/macos_login_items.py

```diff
@@ -0,0 +1,285 @@
+
+        event_data.volume_mount_point = volume_mount_point
+
+    if relative_target_path:
+      event_data.target_path = relative_target_path
+
+    return record_data
+
+  # pylint: disable=arguments-differ
+  def _ParsePlist(self, parser_mediator, top_level=None, **unused_kwargs):
+    """Extracts login item information from the plist.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfVFS.
+      top_level (Optional[dict[str, object]]): plist top-level item.
+    """
+    session_items = top_level.get('SessionItems')
+    if not session_items:
+      return
+
+    for custom_list_item in session_items.get('CustomListItems'):
+      alias_data = custom_list_item.get('Alias')
+      properties = custom_list_item.get('CustomItemProperties', {})
+
+      event_data = MacOSLoginItemEventData()
+      event_data.name = custom_list_item.get('Name')
+      event_data.hidden = properties.get(
+          'com.apple.LSSharedFileList.ItemIsHidden', False)
+
+      if alias_data is not None:
+        self._ParseAliasData(alias_data, event_data)
+
+      parser_mediator.ProduceEventData(event_data)
+
+
+class MacOSLoginItemsPlistPlugin(interface.PlistPlugin):
+  """Base plist parser plugin for Mac OS login items wth bookmark data."""
+
+  # pylint: disable=abstract-method
+
+  def _ConvertHFSDate(self, alias_date):
+    hfs_epoch = dfdatetime_hfs_time.HFSTimeEpoch()
+    hfs_epoch_datetime = datetime(
+      year=hfs_epoch.year,
+      month=hfs_epoch.month,
+      day=hfs_epoch.day_of_month,
+    )
+    hfs_epoch_timedelta = alias_date - hfs_epoch_datetime
+
+    return dfdatetime_hfs_time.HFSTime(
+      timestamp=hfs_epoch_timedelta.total_seconds()
+    )
+
+  def _ParseBookmark(
+      self, parser_mediator, filename, bookmark_index, bookmark_data):
+    """Extracts login item information from bookmark data.
+
+      Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+        and other components, such as storage and dfVFS.
+      filename (str): filename of the plist file being processed.
+      bookmark_index (int): index of the bookmark data in the plist file.
+      bookmark_data (bytes): Mac OS bookmark struct data.
+    """
+    bookmarks = BookmarkParser.parse_bookmark(
+        filename, bookmark_index, '$objects', bookmark_data)
+
+    for bookmark in bookmarks:
+      event_data = MacOSLoginItemEventData()
+      event_data.name = bookmark.get('display_name')
+      event_data.cnid_path = bookmark.get('inode_path')
+      event_data.volume_name = bookmark.get('volume_name')
+      event_data.target_path = bookmark.get('path')
+      event_data.volume_mount_point = bookmark.get('volume_path')
+
+      creation_date = bookmark.get('creation_date')
+      if creation_date is not None:
+        event_data.target_creation_time = self._ConvertHFSDate(creation_date)
```

Comment:
@Spferical bookmark appears to use [Cocoa timestamps](https://dfdatetime.readthedocs.io/en/latest/sources/Date-and-time-values.html#cocoa-timestamp) not [HFS timestamps](https://dfdatetime.readthedocs.io/en/latest/sources/Date-and-time-values.html#hfs-timestamp)